### PR TITLE
Authentication fix

### PR DESCRIPTION
**Pull request by anthonyqzhu**
This PR fixes authentication issue where the oauth flow would never be entered - the /calendar servlet code would run before asking the user for permission resulting in an error. The code now goes through the authorization process no matter what and now works correctly for users who have given permission and also users who have not.

### CODE REVIEW

File: src/main/webapp/grocery.html [hunk 1]
```diff
@@ -12,7 +12,7 @@
                 <button class="dropbtn">MENU</button>
                 <div class="dropdown-content">
                     <a href="home.html">Settings</a>
-                    <a href="cal.html">Calendar</a>
+                    <a href="/calendar-auth">Calendar</a>
```
Comment (yjennywang) [inline src/main/webapp/grocery.html:15]: maybe a naive question, but why /calendar-auth instead of just /calendar?

**Review by yjennywang — APPROVED** [review]
Would be nice to explain what the issue is in the PR description:
"This PR fixes authentication issue where the oauth flow would never be entered"

- when was this happening? how does the change fix it?

File: src/main/webapp/grocery.html (hunk 1 above)
Comment (anthonyqzhu) [reply src/main/webapp/grocery.html:15]: I dont actually think this is necessary, but I was running into an issue where the /calendar servlet code was running before the authorization flow happened, so I decided to make a servlet dedicated to initializing the authorization first which then redirects to cal and then /calendar is called

Comment (anthonyqzhu) [reply src/main/webapp/grocery.html:15]: This can still be changed in the future

**Pull request merged**